v2/git-initializer/main.go

```diff
@@ -0,0 +1,265 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-git/go-billy/v5/osfs"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/cache"
+	"github.com/go-git/go-git/v5/plumbing/transport"
+	gitssh "github.com/go-git/go-git/v5/plumbing/transport/ssh"
+	"github.com/go-git/go-git/v5/storage/filesystem"
+	"github.com/pkg/errors"
+
+	"github.com/urfave/cli/v2"
+
+	"github.com/brigadecore/brigade/sdk/v2/core"
+	"github.com/brigadecore/brigade/v2/internal/signals"
+	"github.com/brigadecore/brigade/v2/internal/version"
+)
+
+// Building w/ CLI frontend such that various opts can be easily overridden
+// for testing, e.g. payload file location, workspace location, etc. (see test.sh)
```

Comment:
I'm still not quite in favor of building this as a CLI. I understand that made it easier to pass options, etc. when invoking this from a suite of bash-based tests, but I don't understand why the tests are bash-based to begin with.

If we were to stick to go-based tests then this program really doesn't need to be a CLI because we can directly test various function invocations per usual _and_ we'd spare ourselves from having to juggle an additional test suite that's implemented in another language.

Reply:
Ok, I've removed the CLI portion and the ported bash tests.  We can work on a golang port of these if we'd like to see a lower, functional level of coverage of this binary.  For now, I've vetted continued working behavior via the e2e tests.

Reply:
That seems like a very fair middle ground.

Want to open an issue to improve unit test coverage on this component? I'll be happy to contribute.

Reply:
👍 https://github.com/brigadecore/brigade/issues/1230